### DISCUSSION TITLE
How to add filter to entity schema ?

### DISCUSSION THREAD

**Discussion by huyhoang160593** [mikro-orm/mikro-orm#4906] [Q&A] [answered]
Can somebody provide me a way to use filter combine with EntitySchema with out using any type of Class, I'm tried to find in the documentation without success

**Comment by B4nan** [comment, marked as answer]
You can always check what the decorator is doing, the code is very simple:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/decorators/Filter.ts

```ts
export const schema = new EntitySchema<IBook>({
  name: 'Book',
  filters: {
    myFilterName: { ...same filter options as you would put into the decorator... },
  },
  properties: {
    id: { type: 'number', primary: true },
    title: { type: 'string' },
  },
});
```

**Comment by huyhoang160593** [comment]
> You can always check what the decorator is doing, the code is very simple:
> 
> https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/decorators/Filter.ts
> 
> ```ts
> export const schema = new EntitySchema<IBook>({
>   name: 'Book',
>   filters: {
>     myFilterName: { ...same filter options as you would put into the decorator... },
>   },
>   properties: {
>     id: { type: 'number', primary: true },
>     title: { type: 'string' },
>   },
> });
> ```
Thanks you for the fast reply, very appreciate that
